<launch>
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find asr_world_model)/log/rosconsole.config" />
    <node pkg="asr_world_model" type="asr_world_model" name="asr_world_model" ns="env" output="screen">
        <rosparam file="$(find asr_world_model)/rsc/world_model_settings.yaml" />
        <rosparam file="$(find asr_world_model)/rsc/world_descriptions/world_description.yaml" />
        <!-- Needed to get bin_size and maxProjectionAngleDeviation -->
        <rosparam file="$(find asr_recognizer_prediction_ism)/rsc/sensitivity.yaml" />
        <!-- Needed to get the dbfilename -->
        <rosparam file="$(find asr_recognizer_prediction_ism)/rsc/scene_recognition.yaml"/>
        <!-- The file name for the itermediate_object weights, generated by the intermediate_object_generator (XXX will be replaced with dbfilename) -->
        <param name="intermediate_object_weight_file_name" value="$(find asr_world_model)/rsc/IntermediateObject_XXX.xml"/>
    </node>
    <node pkg="asr_object_database" type="asr_object_database" name="asr_object_database" output="screen"/>
    <env name="ROSCONSOLE_CONFIG_FILE"
         value="$(find asr_recognizer_prediction_ism)/launch/rosconsole.config"/>
    <node name="rp_ism_node" pkg="asr_recognizer_prediction_ism" type="rp_ism_node" output="screen">
	<rosparam file="$(find asr_recognizer_prediction_ism)/rsc/pose_prediction.yaml"/>
	<rosparam file="$(find asr_recognizer_prediction_ism)/rsc/resampler.yaml"/>
	<rosparam file="$(find asr_recognizer_prediction_ism)/rsc/scene_recognition.yaml"/>
	<rosparam file="$(find asr_recognizer_prediction_ism)/rsc/services.yaml"/>
	<rosparam file="$(find asr_recognizer_prediction_ism)/rsc/shared_memory.yaml"/>
        <rosparam file="$(find asr_recognizer_prediction_ism)/rsc/visualization_dome.yaml"/>
	<rosparam file="$(find asr_ism)/launch/helpers/visualization.yaml"/>
	<rosparam file="$(find asr_ism)/launch/helpers/sensitivity.yaml" />
    </node>
</launch>